crates/oci-distribution/src/lib.rs

```diff
@@ -252,4 +289,24 @@ mod test {
         assert_eq!(manifest.schema_version, 2);
         assert!(!manifest.layers.is_empty());
     }
+
+    #[tokio::test]
+    async fn test_pull_layer() {
+        let image = Reference::try_from(HELLO_IMAGE).expect("parsed reference");
+        let mut c = Client::default();
+        c.auth(&image, None).await.expect("authenticated");
+        let manifest = c.pull_manifest(&image).await.expect("pull manifest");
+
+        // Pull one specific layer
+        let file: Vec<u8> = Vec::new();
```

Comment:
My test is basically writing the data to a vector, where I can run a few trivial tests to make sure the right amount od data was sucked down. (Ignore, for the moment, the network-bound nature of this test.)

Reply:
`Vec<u8>` implements `AsyncWrite` so this should work with `AsyncWrite` instead of `Write`

Reply:
Fixed